### PR TITLE
Further usecases

### PR DESCRIPTION
**Pull request by buenaflor**
This PR includes a multitude of use cases based on the defined layers and features.

### CODE REVIEW

File: doc/usecases/layers/layer_visibility.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  User successfully changes the visibility of layers in their landscape plan by selecting the desired layer from the app's menu and toggling the visibility on or off.
+- **Alternative scenario:**
+  User accidentally selects the wrong image and selects another image to correct the mistake.
+- **Error scenario:**
+  - User attempts to import a file that is not a supported image format or is corrupted and the app displays an error message.
+  - Borders do not close: the app displays an error message.
+- **Postcondition:**
+  The user's selected background image and borders are used for further planning.
+- **Non-functional Constraints:**
+  - Support for multiple image formats
```
Comment (markus2330) [inline doc/usecases/layers/layer_visibility.md:25]: Copy&Paste error?

File: doc/usecases/layers/moisture_layer.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Use Case: Moisture Layer
```
Comment (markus2330) [inline doc/usecases/layers/moisture_layer.md:1]: Please rewrite: moisture is in zones and not elements (like Animals)

File: doc/usecases/layers/paths_layer.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Use Case: Paths Layer
+
+## Summary
+
+- **Scope:** Paths Layer
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** The user can add, edit, move and delete path elements in their plan using the paths layer to represent the movement and flow of traffic in their garden
```
Comment (markus2330) [inline doc/usecases/layers/paths_layer.md:8]:
```suggestion
- **Brief:** The user can add, edit, move and delete a connected network of paths and fences in their plan using the paths layer
```

File: doc/usecases/layers/paths_layer.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+- **Status:** Draft
+
+## Scenarios
+
+- **Precondition:**
+  The user has opened the app and has selected the paths layer.
+- **Main success scenario:**
+  The user successfully adds, edits, moves and deletes path elements in their plan using the paths layer. This includes positioning the path elements in the appropriate location.
```
Comment (markus2330) [inline doc/usecases/layers/paths_layer.md:16]: See above. Also thickness of paths should be select-able (stepping stones, small way, ... roads)

File: doc/usecases/layers/ph_values_layer.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Use Case: pH Values Layer
+
+## Summary
+
+- **Scope:** pH Values Layer
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** The user can add and edit pH values in their plan using the pH values layer
```
Comment (markus2330) [inline doc/usecases/layers/ph_values_layer.md:8]: This is also in zones: drawing with big brush to indicate where the pH value is 0..14.

File: doc/usecases/layers/plants_layer.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+- **Brief:** The user can add, edit, move and delete plant elements in their landscape plan using the plants layer to represent the plants and crops in their garden and view the relationships between the plants through lines connecting them.
+- **Status:** Draft
+
+## Scenarios
+
+- **Precondition:**
+  The user has opened the app and has selected the plants layer.
+- **Main success scenario:**
+  - The user is able to add plants and is presented with a list of plant options
+  - The user positions the plant element in the appropriate location in the plan
```
Comment (markus2330) [inline doc/usecases/layers/plants_layer.md:17]:
```suggestion
  - The user positions the plant element in the appropriate location in the map
```

File: doc/usecases/layers/plants_layer.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  - The user is able to view the relationships between the plants by looking at the lines connecting them. The lines indicate the strength of the relationship
+  - The user is able to move, edit and delete selected plants
```
Comment (markus2330) [inline doc/usecases/layers/plants_layer.md:19]:
```suggestion
  - The user gets a visual indication which parts of the map are better suited for the plant to be placed (e.g. pH value, moisture, animals or shadows).
  - The user is able to move, edit and delete selected plants
```

File: doc/usecases/layers/shadows_layer.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# Use Case: Shadows Layer
+
+## Summary
+
+- **Scope:** Shadow Layer
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** The user can add, edit, move and delete shadow indicators in their plan using the wind layer
```
Comment (markus2330) [inline doc/usecases/layers/shadows_layer.md:8]: - This is again a zones map (big brush showing how shady it is at which places)
- Copy&paste error: wind -> shadow

File: doc/usecases/layers/trees_layer.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# Use Case: Trees Layer
+
+## Summary
+
+- **Scope:** Trees Layer
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** The user can add, edit, move and delete trees and other permanent structures such as bushes in their plan.
```
Comment (markus2330) [inline doc/usecases/layers/trees_layer.md:8]:
```suggestion
- **Brief:** The user can add, edit, move and delete forest, trees and, hedge bushes in their map.
```

File: doc/usecases/layers/trees_layer.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+- **Status:** Draft
+
+## Scenarios
+
+- **Precondition:**
+  The user has opened the app and has selected the trees layer.
+- **Main success scenario:**
+  The user successfully adds, edits, moves and deletes trees and other permanent structures in their landscape plan using the trees layer. This includes positioning the stems, adjusting the height, sketching the shape, and specifying the type of tree or structure.
```
Comment (markus2330) [inline doc/usecases/layers/trees_layer.md:16]: Please put new sentences in new lines. Fix as corrected above.
Forest is an area of trees (individual trees not drawn), hedge an area of bushes.

File: doc/usecases/layers/winds_layer.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
+# Use Case: Winds Layer
+
+## Summary
+
+- **Scope:** Wind Layer
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** The user can add, edit, move and delete wind indicators in their plan using the wind layer
```
Comment (markus2330) [inline doc/usecases/layers/winds_layer.md:8]: Winds is a zones layer (where it is more or less windy)

File: doc/usecases/layers/zones_layer.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
+# Use Case: Zones Layer
+
+## Summary
+
+- **Scope:** Zones Layer
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** The user can add, edit, move and delete zones in their plan using the zones layer. Zones can be added via common shapes or hand drawn polygons.
```
Comment (markus2330) [inline doc/usecases/layers/zones_layer.md:8]: :+1: This is a good idea for zones layers. Please also add brushing functionality, like for other layers.

In zones the zones exclude each other (different to animals, where they can overlap).

Prefer to keep **Brief:** short but details (like how Zones are drawn) to **Main success scenario:**

File: doc/usecases/measuring_distance.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
+# Use Case: Measuring Distance
+
+## Summary
+
+- **Scope:** Measuring Distance
```
Comment (markus2330) [inline doc/usecases/measuring_distance.md:5]:
```suggestion
- **Scope:** All Layers
```

File: doc/usecases/measuring_distance.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** The user can measure the distance between different elements in their plan in order to ensure that the layout is practical and accurate.
```
Comment (markus2330) [inline doc/usecases/measuring_distance.md:8]:
```suggestion
- **Brief:** The user can measure the distance between different elements in their map.
```

File: doc/usecases/measuring_distance.md [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+- **Status:** Draft
+
+## Scenarios
+
+- **Precondition:**
+  The user has opened the app and has selected the desired layer (e.g. plants, paths).
+- **Main success scenario:**
+  - The user selects the "Measure Distance" button.
+  - The user clicks on the first element they want to measure the distance from.
```
Comment (markus2330) [inline doc/usecases/measuring_distance.md:17]:
```suggestion
  - The user clicks on the first point they want to measure the distance from.
```

File: doc/usecases/measuring_distance.md [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+  - The user clicks on the second element they want to measure the distance to.
+  - The app displays the distance between the two elements in the desired unit (e.g. meters, feet).
```
Comment (markus2330) [inline doc/usecases/measuring_distance.md:19]: Please allow any number of points (simply add all distances together)

File: doc/usecases/measuring_distance.md [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+  - The user adjusts the layout of their landscape plan as needed based on the measured distance.
```
Comment (markus2330) [inline doc/usecases/measuring_distance.md:20]: There are several reasons for measuring (e.g. distance between trees)

```suggestion
```

File: doc/usecases/translation.md [hunk 18]
```diff
@@ -0,0 +1,23 @@
+# Use Case: Translation
+
+## Summary
+
+- **Scope:** Translation
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** The user can translate the app's interface and text into English or German
```
Comment (markus2330) [inline doc/usecases/translation.md:8]:
```suggestion
- **Brief:** The user can choose the app's interface to be in English or German
```

File: doc/usecases/translation.md [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+- **Status:** Draft
+
+## Scenarios
+
+- **Precondition:**
+  The user has opened the app and wants to change the language of the interface.
+- **Main success scenario:**
+  The user successfully translates the app's interface and text into English or German by selecting the desired language from the app's menu.
```
Comment (markus2330) [inline doc/usecases/translation.md:16]: Can you additionally guess the language from the browser infos? (But language should be changeable in the app's settings anyway.)

File: doc/usecases/zoom.md [hunk 20]
```diff
@@ -0,0 +1,25 @@
+# Use Case: Zoom
+
+## Summary
+
+- **Scope:** Zoom
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** The user can zoom in and out on their landscape plan to view details more clearly or get a better overview of their garden.
```
Comment (markus2330) [inline doc/usecases/zoom.md:8]:
```suggestion
- **Brief:** The user can zoom in and out on their landscape map to view details more clearly or get a better overview of their map.
```

File: doc/usecases/zoom.md [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+- **Status:** Draft
+
+## Scenarios
+
+- **Precondition:**
+  The user has opened the app and has selected the desired layer (e.g. plants, paths).
+- **Main success scenario:**
+  - The user is able to zoom in on a specific area of the plan.
+  - The user is able to zoom out and view a wider area of the plan.
```
Comment (markus2330) [inline doc/usecases/zoom.md:17]: Can we summarize elements if there are too many at one spot?

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great work! Very thoughtful and many aspects considered! :rocket:

**Comment by markus2330** [comment]
One remark so that we do not get confused with terminology. In my review I wrote "Zones" but it would be better to use the word "Area", to keep the word zones for the "Zones layer". So the zones layers uses areas, like e.g. also the animal layer.

In case of doubt, we can also meet in the next days for discussions/clarifications.

**Comment by buenaflor** [comment]
> Contributor

Got it! Will adjust these changes

File: doc/usecases/translation.md (hunk 19 above)
Comment (buenaflor) [reply doc/usecases/translation.md:16]: Should be possible, will add it to the decision

File: doc/usecases/layers/layer_visibility.md [hunk 22]
```diff
@@ -13,13 +13,11 @@
 - **Precondition:**
   User has opened the app and has multiple layers available to view.
 - **Main success scenario:**
-  User successfully changes the visibility of layers in their landscape plan by selecting the desired layer from the app's menu and toggling the visibility on or off.
+  User successfully changes the visibility of layers in their map by selecting the desired layers and can toggle the visibility on or off.
 - **Alternative scenario:**
-  User accidentally selects the wrong image and selects another image to correct the mistake.
+  User accidentally toggles the visibility of the wrong layer and uses the app's undo function or toggling it again to correct the mistake.
```
Comment (markus2330) [inline doc/usecases/layers/layer_visibility.md:18]: Undo should only apply to real changes, not to changes in visibility.

**Review by markus2330 — APPROVED** [review]
Only very minor things found, can be fixed in another PR.

**Pull request merged**